Sources/BraveWallet/Crypto/Onboarding/RestoreWalletView.swift

```diff
@@ -19,243 +19,190 @@ struct RestoreWalletContainerView: View {
-      case .inputsDontMatch:
-        return Strings.Wallet.passwordsDontMatchError
-      }
-    }
-  }
-
-  @State private var password: String = ""
-  @State private var repeatedPassword: String = ""
-  @State private var phrase: String = ""
-  @State private var isEditingPhrase: Bool = false
-  @State private var showingRecoveryPhase: Bool = false
-  @State private var restoreError: RestoreWalletError?
-  @State private var isShowingLegacyWalletToggle: Bool = false
+  
+  @Environment(\.sizeCategory) private var sizeCategory
+  @Environment(\.dismiss) private var dismiss
+  
   @State private var isBraveLegacyWallet: Bool = false
-
-  private var isBiometricsAvailable: Bool {
-    LAContext().canEvaluatePolicy(.deviceOwnerAuthenticationWithBiometrics, error: nil)
+  @State private var isRevealRecoveryWords: Bool = true
+  @State private var recoveryWords: [String] = .init(repeating: "", count: 12)
+  @State private var newPassword: String?
+  @State private var isShowingCreateNewPassword: Bool = false
+  @State private var isShowingPhraseError: Bool = false
+  @State private var isShowingCompleteState: Bool = false
+  
+  private var numberOfColumns: Int {
+    sizeCategory.isAccessibilityCategory ? 2 : 3
   }
-
-  private func validate(_ completion: @escaping (Bool) -> Void) {
-    keyringStore.isStrongPassword(password) { isValid in
-      if phrase.isEmpty {
-        restoreError = .invalidPhrase
-      } else if !isValid {
-        restoreError = .requirementsNotMet
-      } else if password != repeatedPassword {
-        restoreError = .inputsDontMatch
-      } else {
-        restoreError = nil
-      }
-      completion(restoreError == nil)
-    }
+  
+  private var isLegacyWallet: Bool {
+    recoveryWords.count == 24
   }
-
-  private func restore() {
-    validate { success in
-      if !success {
-        return
-      }
-      keyringStore.restoreWallet(
-        phrase: phrase,
-        password: password,
-        isLegacyBraveWallet: isBraveLegacyWallet
-      ) { success in
-        if !success {
-          restoreError = .invalidPhrase
-        } else {
-          keyringStore.resetKeychainStoredPassword()
-          if isBiometricsAvailable {
-            keyringStore.isRestoreFromUnlockBiometricsPromptVisible = true
-          } else {
-            // If we're displaying this via onboarding, mark as completed.
-            keyringStore.markOnboardingCompleted()
-          }
-        }
-      }
-    }
-  }
-
-  private func handlePhraseChanged(_ value: String) {
-    isShowingLegacyWalletToggle = value.split(separator: " ").count == 24
-    if restoreError == .invalidPhrase {
-      // Reset validation on user changing
-      restoreError = nil
+  
+  private var isContinueDisabled: Bool {
+    var wordCount = 0
+    for word in recoveryWords where !word.isEmpty {
+      wordCount += 1
     }
+    return wordCount != (isLegacyWallet ? .legacyWalletRecoveryPhraseNumber : .regularWalletRecoveryPhraseNumber)
   }
```

Comment:
```suggestion
  private var isContinueDisabled: Bool {
    recoveryWords.allSatisfy({ !$0.isEmpty })
  }
```

nit: I believe you can just use `allSatisfy` here because `recoveryWords` array will either be updated to the correct length when switching from 12/24 words 🙂